haptic_control:
  ros__parameters:
    channel: "SimpleChannelUDP"
    ff_device_ip_address: "192.168.100.53"
    ff_device_param_file: "/etc/Haption/Connector/desktop_6D_n65.param"
    local_ip_address: "192.168.100.50"
    haptic_control_rate: 1000.0
    max_force: 6.0 # Newton

    enable_safety_sphere: false
    safety_sphere_radius: 0.48
    min_x:  -100.0
    max_x:  100.0
    min_y:  -100.0
    max_y:  100.0
    min_z:  -100.0
    max_z:  100.0
    # min_x_:  0.33
    # max_x_:  0.48
    # min_y_: -0.03
    # max_y_:  0.18
    # min_z_:  -0.10
    # max_z_:  0.13


    delay: 0.0 # round trip simulated delay in seconds (0.0 for no delay)
    force_scale : 0.3
    # safety box that can be activated (centered in the current ee position) -> Not implemented yet
    enable_safety_box: false 
    safety_box_length: 0.2
    safety_box_width: 0.2
    safety_box_height: 0.2

    # vf parameters:
    mesh_type: "file" # knot, sphere, file, bunny 
    output_mesh_path : "/home/nardi/virtual_fixture_vis.obj" # path to save the output mesh for Rviz visualization
    input_mesh_path: "/home/nardi/SKEL_WS/ros2_ws/final_vf.obj" # path to the input mesh if mesh_type is file
    skin_mesh_path: "/home/nardi/SKEL_WS/ros2_ws/skin_mesh.obj" # path to the skin mesh if mesh_type is file
    tool_radius: 0.002 # radius of the end effector sphere
    tool_radius_visualization: 0.005 # radius of the end effector sphere for visualization in Rviz
    
    lookup_area: 0.005 # area around the end effector to look for the closest point on the mesh
    plane_size: 0.002 # plane size used to visualize active constraints in rviz